src/subjects/subjects.controller.ts

```diff
@@ -2,63 +2,77 @@ import {
+import { CreateSubjectDTO } from "./dto/create-subject.dto";
+import { UpdateSubjectDTO } from "./dto/update-subject.dto";
+import { HasRightsGuard } from "./guards/has-rights.guard";
+import { SubjectsService } from "./subjects.service";
+import { YearCourseParams } from "./params/YearCourseParams";
 
 @ApiBearerAuth()
-@ApiTags('subjects')
+@ApiTags("subjects")
 @Controller()
 export class SubjectsController {
-  constructor(private subjectsService: SubjectsService, private yearCourseService: YearCoursesService) {}
+  constructor(
+    private subjectsService: SubjectsService,
+    private yearCourseService: YearCoursesService
+  ) {}
 
   @UseGuards(JwtAuthGuard)
-  @Get('subjects')
+  @Get("subjects")
   async findAll(): Promise<Subject[]> {
     return this.subjectsService.findAll();
   }
 
   @UseGuards(JwtAuthGuard, HasRightsGuard)
-  @Get('yearCourses/:yearCourseId/subjects')
-  async findByYearCourse(@Param('yearCourseId') yearCourseId): Promise<Subject[]> {
-    const yearCourse = await this.yearCourseService.findById(yearCourseId);
+  @Get("yearCourses/:yearCourseId/subjects")
+  async findByYearCourse(
+    @Param() params: YearCourseParams
+  ): Promise<Subject[]> {
+    const yearCourse = await this.yearCourseService.findById(params.id);
 
     return this.subjectsService.findByYearCourse(yearCourse);
   }
 
   @UseGuards(JwtAuthGuard, HasRightsGuard)
-  @Post('yearCourses/:yearCourseId/subjects')
-  async create(@Param('yearCourseId') yearCourseId, @Body() body: CreateSubjectDTO): Promise<Subject> {
-    const yearCourse = await this.yearCourseService.findById(yearCourseId);
+  @Post("yearCourses/:yearCourseId/subjects")
+  async create(
+    @Param() params: YearCourseParams,
+    @Body() body: CreateSubjectDTO
+  ): Promise<Subject> {
+    const yearCourse = await this.yearCourseService.findById(params.id);
 
     return this.subjectsService.create(body.name, body.shortName, yearCourse);
   }
 
   @UseGuards(JwtAuthGuard, HasRightsGuard)
-  @Put('yearCourses/:yearCourseId/subjects')
-  async update(@Param('yearCourseId') yearCourseId, @Body() body: UpdateSubjectDTO): Promise<Subject> {
-    const yearCourse = await this.yearCourseService.findById(yearCourseId);
-    return this.subjectsService.update(body.id, body.name, body.shortName, yearCourse);
+  @Put("yearCourses/:yearCourseId/subjects")
+  async update(
+    @Param() params: YearCourseParams,
+    @Body() body: UpdateSubjectDTO
+  ): Promise<Subject> {
+    const yearCourse = await this.yearCourseService.findById(params.id);
+    return this.subjectsService.update(
+      body.id,
+      body.name,
+      body.shortName,
+      yearCourse
+    );
   }
 
+  // todo: wtf is :yearCourseId and :id
   @UseGuards(JwtAuthGuard, HasRightsGuard)
-  @Delete('yearCourses/:yearCourseId/subjects/:id')
-  async remove(@Param('id') id): Promise<Subject> {
-    return this.subjectsService.remove(id);
+  @Delete("yearCourses/:yearCourseId/subjects/:id")
```

Comment:
Missing definition for `:id`. Params should implement interface which defines both `yearCourseId` and `subjectId`